Formula/libxml2.rb

```diff
@@ -41,7 +41,7 @@ class Libxml2 < Formula
 
   keg_only :provided_by_osx
 
-  depends_on :python => :optional
+  depends_on :python if MacOS.version <= :snow_leopard
```

Comment:
>The situation for formulas like vim which embed a Python interpreter is very different

@tdsmith can you clarify what you think the situation for formulae like vim should be?